checks.d/yarn.py

```diff
@@ -0,0 +1,264 @@
+    'availableVirtualCores': ('yarn.metrics.availableVirtualCores', GAUGE),
+    'allocatedVirtualCores': ('yarn.metrics.allocatedVirtualCores', GAUGE),
+    'totalVirtualCores': ('yarn.metrics.totalVirtualCores', GAUGE),
+    'containersAllocated': ('yarn.metrics.containersAllocated', GAUGE),
+    'containersReserved': ('yarn.metrics.containersReserved', GAUGE),
+    'containersPending': ('yarn.metrics.containersPending', GAUGE),
+    'totalNodes': ('yarn.metrics.totalNodes', GAUGE),
+    'activeNodes': ('yarn.metrics.activeNodes', GAUGE),
+    'lostNodes': ('yarn.metrics.lostNodes', GAUGE),
+    'unhealthyNodes': ('yarn.metrics.unhealthyNodes', GAUGE),
+    'decommissionedNodes': ('yarn.metrics.decommissionedNodes', GAUGE),
+    'rebootedNodes': ('yarn.metrics.rebootedNodes', GAUGE),
+}
+
+# Application metrics for YARN
+YARN_APP_METRICS = {
+    'progress': ('yarn.apps.progress', GAUGE),
+    'startedTime': ('yarn.apps.startedTime', GAUGE),
+    'finishedTime': ('yarn.apps.finishedTime', GAUGE),
+    'elapsedTime': ('yarn.apps.elapsedTime', GAUGE),
+    'allocatedMB': ('yarn.apps.allocatedMB', GAUGE),
+    'allocatedVCores': ('yarn.apps.allocatedVCores', GAUGE),
+    'runningContainers': ('yarn.apps.runningContainers', GAUGE),
+    'memorySeconds': ('yarn.apps.memorySeconds', GAUGE),
+    'vcoreSeconds': ('yarn.apps.vcoreSeconds', GAUGE),
+}
+
+YARN_NODE_METRICS = {
+    'lastHealthUpdate': ('yarn.node.lastHealthUpdate', GAUGE),
+    'usedMemoryMB': ('yarn.node.usedMemoryMB', GAUGE),
+    'availMemoryMB': ('yarn.node.availMemoryMB', GAUGE),
+    'usedVirtualCores': ('yarn.node.usedVirtualCores', GAUGE),
+    'availableVirtualCores': ('yarn.node.availableVirtualCores', GAUGE),
+    'numContainers': ('yarn.node.numContainers', GAUGE),
+}
+
+
+class YarnCheck(AgentCheck):
+
+    def check(self, instance):
+
+        # Build the resourceManager URI
+        resourcemanager_hostname = instance.get('resourcemanager_address', DEFAULT_HOSTNAME)
+        resourcemanager_port = instance.get('resourcemanager_port', DEFAULT_RESOURCEMANAGER_PORT)
+        rm_address = resourcemanager_hostname + ':' + str(resourcemanager_port)
+
+        # Get the cluster ID from Yarn
+        cluster_id = self._get_cluster_id(rm_address)
+
+        # If a Cluster ID was not found, assign it 0
+        if not cluster_id:
+            cluster_id = 0
+
+        self._yarn_cluster_metrics(resourcemanager_hostname, cluster_id, rm_address)
+        self._yarn_app_metrics(resourcemanager_hostname, rm_address)
+        self._yarn_node_metrics(resourcemanager_hostname, cluster_id, rm_address)
+
+    def _get_cluster_id(self, rm_address):
+
+        # Get the cluster info
+        info_json = self._rest_request_to_json(rm_address, INFO_URI)
+
+        if info_json:
+            if info_json['clusterInfo']:
+                if info_json['clusterInfo']['id']:
+                    return info_json['clusterInfo']['id']
+
```

Comment:
Thanks @zachradtka 
I think it would be better to have:
```python
cluster_id = info_json.get("clusterInfo", {}).get("id")
if cluster_id is not None:
   ...
```
instead of all the if statements. Also this would ensure that we wouldn't get a KeyError